File: 2024_java_basics/14-nested-loops-lab/tasks/01-number-pyramid.java
import java.util.Scanner;

public class NumberPyramid {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        
        int number = 1;
        boolean isBigger = false;
        
        for (int rows = 1; rows <= n; rows++) {
            for (int columns = 1; columns <= rows; columns++) {
                if (number > n) {
                    isBigger = true;
                    break;
                }

                System.out.print(number + " ");
                number++;
            }

            if (isBigger) {
                break;
            }

            System.out.println();
        }
        
        scanner.close();
    }
}

File: 2022_c#_basics/14-nested-loops-exercise/01-number-pyramid.cs
using System;

namespace _01._Number_Pyramid
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());

            int number = 1;
            bool isBigger = false;

            for (int rows = 1; rows <= n; rows++)
            {
                for (int colums = 1; colums <= rows; colums++)
                {
                    if (number > n)
                    {
                        isBigger = true;
                        break;
                    }

                    Console.Write(number + " ");
                    number++;
                }

                if (isBigger)
                {
                    break;
                }

                Console.WriteLine();
            }
        }
    }
}
